src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/AdmissionControlAction.java

```diff
@@ -0,0 +1,227 @@
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions;
+
+import static com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Dimension.ADMISSION_CONTROL;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.AppContext;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.RcaConf;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.NodeKey;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.annotations.SerializedName;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class AdmissionControlAction extends SuppressibleAction {
+
+    public static final String NAME = "AdmissionControlAction";
+
+    protected static final long DEFAULT_COOL_OFF_PERIOD_IN_MILLIS = TimeUnit.SECONDS.toMillis(30);
+
+    private final NodeKey esNode;
+    private final String controllerName;
+    private final boolean canUpdate;
+    private final double desiredValue;
+    private final double currentValue;
+
+    public AdmissionControlAction(
+            AppContext appContext,
+            final NodeKey esNode,
+            final String controllerName,
+            final boolean canUpdate,
+            final double desiredValue,
+            final double currentValue) {
+        super(appContext);
+        this.esNode = esNode;
+        this.canUpdate = canUpdate;
+        this.desiredValue = desiredValue;
+        this.currentValue = currentValue;
+        this.controllerName = controllerName;
+    }
+
+    public static Builder newBuilder(
+            final NodeKey esNode,
+            final String controllerName,
+            final AppContext appContext,
+            final RcaConf conf) {
+        return new Builder(esNode, controllerName, appContext, conf);
+    }
+
+    public String getControllerName() {
+        return controllerName;
+    }
+
+    @Override
+    public boolean canUpdate() {
+        return this.canUpdate;
+    }
+
+    @Override
+    public long coolOffPeriodInMillis() {
+        return DEFAULT_COOL_OFF_PERIOD_IN_MILLIS;
+    }
+
+    @Override
+    public String name() {
+        return NAME;
+    }
+
+    @Override
+    public List<NodeKey> impactedNodes() {
+        return Collections.singletonList(esNode);
+    }
+
+    @Override
+    public Map<NodeKey, ImpactVector> impact() {
+        final ImpactVector impactVector = new ImpactVector();
+        if (desiredValue > currentValue) {
+            impactVector.increasesPressure(ADMISSION_CONTROL);
+        } else if (desiredValue < currentValue) {
+            impactVector.decreasesPressure(ADMISSION_CONTROL);
+        }
+        return Collections.singletonMap(esNode, impactVector);
+    }
```

Comment:
ImpactVector `increasesPressure` and `decreasesPressure`, both directions have an action associated to them. Will we decrease the admission control entry param value is the impactVector has `decreasePressure`?

Reply:
AdmissionControl tuning is bi-directional. We mark increase/decrease pressure based on desired value. Added same in code comments.